apps/system/test/unit/task_manager_test.js

```diff
@@ -382,52 +309,58 @@ suite('system/TaskManager >', function() {
                           scrollToDescriptor);
 
     screenNode.parentNode.removeChild(screenNode);
-    navigator.mozSettings = realMozSettings;
-    window.SettingsListener = realSettingsListener;
     navigator.mozL10n = realL10n;
   });
 
-  // The whole suite should use fakeTimers to prevent intemittents
-  // since the code logic is timer-heavy
   setup(function() {
+    apps = {};
+    MockStackManager.mStack = [];
     MockService.mockQueryWith('getHomescreen', home);
     MockService.mockQueryWith('fetchCurrentOrientation', 'portrait-primary');
     MockService.mockQueryWith('defaultOrientation', 'portrait-primary');
+    // The whole suite should use fakeTimers to prevent intemittents
+    // since the code logic is timer-heavy
     this.sinon.useFakeTimers();
 
     scrollProperties['cards-view'] = {
       scrollLeft: 0, scrollTop: 0
     };
+    // reset DOM
+    cardsList.innerHTML = '';
```

Comment:
again, all for having this in a general teardown, but also having it here seems weird